Feature/chatmessage api

Dayeon-00 opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #213
- 작업 요약: 채팅 메시지 내역 조회.

## 📄 상세 내용
- [x] CursorPagination 적용 (페이지 당 100개의 메세지 조회)
- [x] 해당 그룹 채팅방에 접근할 권한이 있는지 확인하는 IsMemberOnStudyGroup 권한추가
- [x] 그룹 채팅방에 메시지 내역을 CursorPagination을 통해 조회할 수 있는 ChatMessageListView 추가
- [x] `api/v1/chat/room/<uuid: study_group_id>/messages 엔드포인트에 ChatMessageListView 매핑

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Dayeon-00 committed:
chat message 상세 조회 testcode 작성

Dayeon-00 committed:
chat message 상세 조회 testcode 작성

Meoyoug commented on apps/chat/serializers.py:
```diff
@@ -28,3 +30,85 @@ class ChatRoomSerializer(serializers.ModelSerializer[StudyGroup]):
+            "uuid": {"read_only": True}
+        }
+
+    def get_profile_img_url(self, obj: User) -> str:
```
프로필 이미지가있다면 프로필 이미지를 내려줘야하는데 리턴값이 똑같네요?

Meoyoug commented on apps/chat/serializers.py:
```diff
@@ -28,3 +30,85 @@ class ChatRoomSerializer(serializers.ModelSerializer[StudyGroup]):
+        models = ChatMessage
+        fields = ["message_id", "sender", "content", "create_at"]
+
+class ChatMessageDetailSerializer(serializers.ModelSerializer[ChatMessage]):
```
이 시리얼라이저는 사용되지 않는 것 같으니 제거해주세요

Meoyoug commented on apps/chat/serializers.py:
```diff
@@ -28,3 +30,85 @@ class ChatRoomSerializer(serializers.ModelSerializer[StudyGroup]):
+
+class ChatMessageDetailSerializer(serializers.ModelSerializer[ChatMessage]):
+    sender = SenderInfoSerializer(read_only=True)
+    time_display = serializers.SerializerMethodField()
```
여기부터 아래의 필드가 필요한지 생각해보셔야 할 것 같습니다! 와이어프레임 참고하시면 좋을 것 같아요

Meoyoug commented on apps/chat/views.py:
```diff
@@ -46,3 +45,94 @@ def get(self, request: Request) -> Response:
+        )
+        except StudyGroup.DoesNotExist:
+            return Response(
+                {"error": "채팅방을 찾을 수 없거나 접근 권한이 없습니다."},
```
찾을 수 없음과 접근 권한이 없음은 분리해주셔야 할 것 같습니다

Meoyoug commented on apps/chat/views.py:
```diff
@@ -46,3 +45,94 @@ def get(self, request: Request) -> Response:
+                # 오래된 이전 메시지들
+                queryset = queryset.filter(id__lt=cursor_id)
+            except (ValueError, IndexError):
+                pass
```
잘못된 커서를 입력했다면 api 스펙상 에러응답을 발생시키는게 더좋아보이네요
무한스크롤을 위한 API인데 커서값이 잘못보내어진 경우에 기본값을 내뱉으면 데이터 중복 노출등의 이슈가 있을 것 같아요

Meoyoug reviewed (changes requested):
커서페이지네이션을 직접 구현하셨는데 DRF 의 CursorPagination 객체도 있으니 활용해보시면 좋을 것 같아요

Dayeon-00 committed:
오류 수정

Dayeon-00 committed:
✨ feat: pagination 기능 추가

permission class를 통해 인증받지 않은 사용자 검증

Dayeon-00 committed:
✨ feat: pagination 기능 추가

첫 진입시 300개 조회하고, 무한스크롤시 100개 조회하는 기능

Dayeon-00 committed:
🐛 fix      : 버그 수정

api 문서에 맞도록 응답 수정 및 isort, black을 했음.

Dayeon-00 committed:
✨ feat: 매시지 조회 기능 추가
🐛 fix: API 응답대로 수정
✅ test: 채팅방 목록 조회 API 테스트 수정 및 메시지 조회 API 테스트 추가

채팅방 목록 조회 API 테스트 수정 및 메시지 조회 API 테스트 추가

Dayeon-00 committed:
🐛 fix: url 오류 수정

urls.py에서 생긴 오류를 수정하였음.

Dayeon-00 committed:
✨ feat: 메세지 조회 기능 view 추가
🐛 fix: API 명세서의 응답대로 수정 및 500 에러를 수정하였음.

view에서 타입 에러가 발생한 것 수정, mypy 문제 수정, 500 에러 수정, 메세지 조회 기능 view 추가
